packages/renderer/src/lib/kube/pods/PodDetails.svelte

```diff
@@ -0,0 +1,101 @@
+<script lang="ts">
+import type { V1Pod } from '@kubernetes/client-node';
+import { StatusIcon, Tab } from '@podman-desktop/ui-svelte';
+import { onMount } from 'svelte';
+import { router } from 'tinro';
+import { stringify } from 'yaml';
+
+import { kubernetesCurrentContextPods } from '/@/stores/kubernetes-contexts-state';
+
+import Route from '../../../Route.svelte';
+import MonacoEditor from '../../editor/MonacoEditor.svelte';
+import PodIcon from '../../images/PodIcon.svelte';
+import DetailsPage from '../../ui/DetailsPage.svelte';
+import StateChange from '../../ui/StateChange.svelte';
+import { getTabUrl, isTabSelected } from '../../ui/Util';
+import KubeEditYaml from '../KubeEditYAML.svelte';
+import KubernetesTerminalBrowser from './KubernetesTerminalBrowser.svelte';
+import { PodUtils } from './pod-utils';
+import PodActions from './PodActions.svelte';
+import PodDetailsLogs from './PodDetailsLogs.svelte';
+import PodDetailsSummary from './PodDetailsSummary.svelte';
+import type { PodUI } from './PodUI';
+
+interface Props {
+  name: string;
+  namespace: string;
+}
+let { name, namespace }: Props = $props();
+
+let pod = $state<PodUI>();
+let detailsPage = $state<DetailsPage>();
+let kubePod = $state<V1Pod | undefined>();
+let kubeError = $state<string>();
+
+onMount(() => {
+  const podUtils = new PodUtils();
+  // loading pod info
+  return kubernetesCurrentContextPods.subscribe(pods => {
+    const matchingPod = pods.find(pod => pod.metadata?.name === name && pod.metadata?.namespace === namespace);
+    if (matchingPod) {
+      try {
+        pod = podUtils.getPodUI(matchingPod);
+        loadDetails().catch((err: unknown) => console.error(`Error getting pod details ${pod?.name}`, err));
+      } catch (err) {
+        console.error(err);
+      }
+    } else if (detailsPage) {
+      // the pod has been deleted
+      detailsPage.close();
```

Comment:
```suggestion
      detailsPage?.close();
```

This is problematic, the `DetailsPage` is bellow in a if statement, so this is nullable, and can cause error.

Reply:
Fixed in new commit.